pandas/_libs/tslib.pyx

```diff
@@ -514,7 +514,7 @@ cpdef array_to_datetime(
                         found_tz = True
                         if utc_convert:
                             _ts = convert_datetime_to_tsobject(val, None)
-                            iresult[i] = _ts.value
+                            iresult[i] = Timestamp(val)._as_unit("ns").value
```

Comment:
Currently there's no op to convert a _TSObject value to another unit without going through the `Timestamp` constructor?

Reply:
i don't think that methods exists, but i was thinking of implementing it anyway, so now seems like a good time